### PR TITLE
feat: Add no artworks have insights Message

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [CX-2597]

### Description

This PR adds a new message in case the user has artworks but no market insights. I refactored a lot of other files as well. The interesting changes can be found in [MyCollectionInsights](https://github.com/artsy/eigen/pull/6808/files#diff-8530431ffcc300fd1acd13bcf0cc7749dc47b0d50b1cb20e440e6929c6af65af) and in [Messages](https://github.com/artsy/eigen/pull/6808/files#diff-b7978becb472eac134c3d686f73395a537e52e5e6e3e1d813c95c7989d791d73).

Simulator Screen Shot - iPhone 12 Pro Max - 2022-05-30 at 16 44 31

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-2597]: https://artsyproduct.atlassian.net/browse/CX-2597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 1]
```diff
@@ -91,85 +93,69 @@ const MyCollection: React.FC<{
   // hack for tests. we should fix that.
   const setJSX = useContext(StickyTabPageFlatListContext).setJSX
 
-  const space = useSpace()
-  const toast = useToast()
+  const showMessages = async () => {
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:96]: Just refactoring in this file to clean up 🧹

File: src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsForArtistsYouCollectRail.tsx [hunk 2]
```diff
@@ -10,15 +10,15 @@ import { graphql, useFragment } from "react-relay"
 import { useScreenDimensions } from "shared/hooks"
 
 interface AuctionResultsForArtistsYouCollectRailProps {
-  auctionResults: AuctionResultsForArtistsYouCollectRail_me$key
+  me: AuctionResultsForArtistsYouCollectRail_me$key
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsForArtistsYouCollectRail.tsx:13]: Cleaning up 🧹

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 3]
```diff
@@ -103,10 +127,11 @@ export const MyCollectionInsightsScreenQuery = graphql`
   query MyCollectionInsightsQuery {
     me {
       ...AuctionResultsForArtistsYouCollectRail_me
-      auctionResults: myCollectionAuctionResults(first: 1) {
+      auctionResults: myCollectionAuctionResults(first: 3) {
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:130]: This didn't work with `1` and always returned `artworksCount: 1`  🤷

**Review by MrSltun — APPROVED** [review]
LGTM! Great work Ole! 👏 🌟

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:130]: interesting, was it also returning 1 when there are no artworks at all? here I just wanted to know that there is at least 1 auction result available

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
+    />
+  )
+}
+
+export const ArtworksHaveNoInsights: React.FC<Partial<MessageProps>> = (props) => {
+  return (
+    <Message
+      variant="info"
+      title="Why don’t I see insights for all my artworks?"
+      text="Insights are market data on artists you collect. Not all artists in your collection currently have insights available."
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:20]: **issue:** I think in the case of no insights available, we should return `There are currently no insights available on your collection. Insights will be shown here if they become available.` and as a title it should be `New artwork successfully added`

Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:20]: see https://www.figma.com/file/l3plkt7gMSg4p1Or0E8SLC/My-Collection-2022?node-id=4422%3A87507

**Review by MounirDhahri — CHANGES_REQUESTED** [review]

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx (hunk 3 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:130]: Exactly, this was the problem. TotalCount was returning 1 for some reason. I'll look into Metaphysics as well or even better expose a field to check if there are any insights.

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx (hunk 4 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:20]: This PR is implementing the following message: https://www.figma.com/file/l3plkt7gMSg4p1Or0E8SLC/My-Collection-2022?node-id=4258%3A77527

Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:20]: It's displayed if there are artworks uploaded already but none of these artworks have insights.

Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:20]: I thought that this message is what we want to show when we have partial insights

**Pull request merged**